### PR TITLE
Offset `Specimen_collection_date_time` by device test length

### LINKED ISSUE #2607: Offset specimen collection date using device timer
### Description
Per the thread with Ohio on result date and specimen collect date. Since all of our tests are rapid test we had set both fields to the time when the result was submitted. This was fine when were were only sending the date. It looks like we added the time in for CA https://github.com/CDCgov/prime-simplereport/issues/2440. This is probably why it recently stopped working for Ohio

For a quick work around I think we should subtract the timer time from the result date and report that as the specimen collect date. Example if I submit a test result in SR at 11:30 AM the specimen collection date would be 11:15AM

## Acceptance Criteria
- When serializing TestEventExport for ReportStream, subtract the testing device's timer length from its test_date for specimen collection date

### PR DESCRIPTION
**Pull request by nullflux**
## Related Issue or Background Info

- Resolves #2607 

## Changes Proposed

- See title

## Additional Information

- We're supposed to "test this" somehow, with the state? Will coordinate w/ ReportStream

### UI
- [x] Any changes to the UI/UX are approved by design 
- [x] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [x] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [x] Database changes are submitted as a separate PR
  - [x] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [x] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [x] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
  - [x] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [x] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [x] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [x] Any dependencies introduced have been vetted and discussed

## Cloud
- [x] DevOps team has been notified if PR requires ops support
- [x] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

File: backend/src/test/java/gov/cdc/usds/simplereport/api/model/TestEventExportTest.java [hunk 1]
```diff
@@ -145,4 +147,22 @@ void determinesFirstTestFromTestEvent_hasPriorTest_null() {
...
+    Person p = _dataFactory.createFullPerson(o);
+    TestEvent te = _dataFactory.createTestEvent(p, f, TestResult.NEGATIVE, null);
+
+    var testLength = f.getDefaultDeviceType().getTestLength();
+    var specimenTime =
+        Date.from(te.getDateTested().toInstant().plus(Duration.ofMinutes(testLength)));
+    TestEventExport sut = new TestEventExport(te);
+
+    assertEquals(
+        TestEventExport.dateToHealthCareString(
```
Comment (zdeveloper) [inline backend/src/test/java/gov/cdc/usds/simplereport/api/model/TestEventExportTest.java:164]: This test uses the same logic as the implementation for the expected, meaning any flaw in the production code will also change the test expectation, thus leading to test to pass. I recommend hard-coding the expectation. and reverting the static methods to private again.

File: backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java [hunk 2]
```diff
@@ -35,6 +37,7 @@
  */
 public class TestEventExport {
   public static final String CSV_API_VERSION = "05Aug2021"; // last time we changed something
+  private static final int FALLBACK_DEFAULT_TEST_LENGTH = 15;
```
Comment (zdeveloper) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java:40]: perhaps include _MINUTES in the name to include the unit.

Comment (nullflux) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/model/TestEventExport.java]: Updated in d84a2fc3

File: backend/src/test/java/gov/cdc/usds/simplereport/api/model/TestEventExportTest.java (hunk 1 above)
Comment (nullflux) [reply backend/src/test/java/gov/cdc/usds/simplereport/api/model/TestEventExportTest.java]: Resolved in 3c43bf37

**Review by zdeveloper — APPROVED** [review]
LGTM

**Pull request merged**